cmd/release-controller/http.go

```diff
@@ -1230,3 +1258,26 @@ var extendedRelTime = []humanize.RelTimeMagnitude{
 func relTime(a, b time.Time, albl, blbl string) string {
 	return humanize.CustomRelTime(a, b, albl, blbl, extendedRelTime)
 }
+
+type releaseInfoStub struct {
+	Metadata *struct {
+		Version  string   `json:"version"`
+		Previous []string `json:"previous"`
+	} `json:"metadata"`
+}
+
+func (c *Controller) getSupportedUpgrades(tagPull string) ([]string, error) {
+	var tagReleaseInfo releaseInfoStub
+	tagReleaseJson, err := c.releaseInfo.ReleaseInfo(tagPull)
```

Comment:
The API here is wrong, ReleaseInfo should be what is returned by this API, not JSON.  Move the struct into that file, return it from the interface, and define the struct as public (same as the other apis).

Reply:
I have reworked all of this logic.  Please take another look.